src/framework/stdext/math.cpp

```diff
@@ -62,4 +63,39 @@ namespace stdext
         static std::uniform_real_distribution<float> dis(0.0, 1.0);
         return min + (max - min) * dis(gen);
     }
-}
\ No newline at end of file
+
+    std::mt19937& random_gen()
+    {
+	    static std::random_device rd;
+	    static std::mt19937 generator(rd());
+	    return generator;
+    }
+
+    bool random_bool(double probability)
+    {
+	    static std::bernoulli_distribution booleanRand;
+	    return booleanRand(random_gen(), std::bernoulli_distribution::param_type(probability));
+    }
+
+    int32_t normal_random(int32_t minNumber, int32_t maxNumber)
+    {
+	    static std::normal_distribution<float> normalRand(0.5f, 0.25f);
+	    if (minNumber == maxNumber) {
+		    return minNumber;
+	    } else if (minNumber > maxNumber) {
+		    std::swap(minNumber, maxNumber);
+	    }
+
+	    int32_t increment;
+	    const int32_t diff = maxNumber - minNumber;
+	    const float v = normalRand(random_gen());
+	    if (v < 0.0) {
+		    increment = diff / 2;
+	    } else if (v > 1.0) {
+		    increment = (diff + 1) / 2;
+	    } else {
+		    increment = round(v * diff);
+	    }
+	    return minNumber + increment;
+    }
```

Comment:
There is a new version of this function in TFS that has fixed some issues
https://github.com/otland/forgottenserver/commit/16a985c33c80544af0ac0668bc35da11cfcaf9f9